### ISSUE TITLE
Bash completions don't work

### ISSUE THREAD

**Issue by sealor** [rust-lang/cargo#12409]
### Problem

The Bash completions don't work because they are not be sourced correctly.

### Steps

Rustup expects Bash completions in `~/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/etc/bash_completion.d/cargo`:
```
$ rustup completions bash cargo
if command -v rustc >/dev/null 2>&1; then
        source "$(rustc --print sysroot)"/etc/bash_completion.d/cargo
fi
```

Unfortunately, the completions are not there:
```
$ ls -d "$(rustc --print sysroot)"/etc/bash_completion.d/cargo
ls: cannot access '/home/stefan/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/etc/bash_completion.d/cargo': No such file or directory
```

But, the completions are here:
```
$ file "$(rustc --print sysroot)"/src/etc/bash_completion.d/cargo
/home/stefan/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/src/etc/bash_completion.d/cargo: ASCII text
```
Note the `/src/etc` instead of just `/etc`.

### Possible Solution(s)

The Bash completions should be placed in `/etc/bash_completion.d/cargo` instead of `/src/etc/bash_completion.d/cargo`.

### Version

```text
$ cargo --version
cargo 1.71.0 (cfd3bbd8f 2023-06-08)
```

**Comment by weihanglo** [comment]
Thanks for the report, and sorry for the inconvenience. This is a regression and planned to get a stable backport soon. See <https://github.com/rust-lang/rustup/issues/3407>. Will keep this issue open till a new point release comes out (presumbly 1.71.1).

@rustbog label -S-triage +S-blocked-external +A-completions

**Comment by weihanglo** [comment]
Fixed in 1.71.1 with <https://github.com/rust-lang/rust/pull/113579>. Thanks you for reporting this!
